### PR TITLE
Add Kleisli arrow specialized to Either, with monoid instance.

### PR DESCRIPTION
**Pull request by nuttycom**
Review by @andywhite37

### CODE REVIEW

File: src/thx/Eithers.hx [hunk 1]
```diff
@@ -156,3 +157,41 @@ class Eithers {
...
+
+  public function ap<R0>(e: EitherK<A, L, R -> R0>): EitherK<A, L, R0> {
+    return flatMap(function(r: R) return e.map(function(f: R -> R0) return f(r)));
+  }
+
+  public function flatMap<R0>(f: R -> EitherK<A, L, R0>): EitherK<A, L, R0> {
+    return function(a: A) return Eithers.flatMap(this(a), function(r: R) return f(r).apply(a));
+  }
+
+  public static function monoid<L, R>(): Monoid<EitherK<R, L, R>> {
```
Comment (nuttycom) [inline src/thx/Eithers.hx]: A monad is just a monoid in the category of endofunctors. 😀

Comment (andywhite37) [inline src/thx/Eithers.hx]: obviously

**Comment by mlms13** [comment]
:shipit:

**Comment by andywhite37** [comment]
My only comment is questioning whether this new type `EitherK` belongs in Eithers.hx, or its own file?

**Pull request merged**